src/blocks/ImageBlock/ImageBlock.test.ts

```diff
@@ -0,0 +1,61 @@
+import { renderToFragment } from '@lib/renderer';
+import { describe, expect, test } from 'vitest';
+import ImageBlock, { type Props } from './ImageBlock.astro';
+
+
+
+describe('ImageBlock', () => {
+  test('Block renders image', async () => {
+    const fragment = await renderToFragment<Props>(ImageBlock, {
+      props: {
+        block: {
+          id: '123',
+          image: {
+            url: 'https://example.com/',
+            alt: 'A test image',
+            title: 'A test image',
+            height: 150,
+            width: 150
+          }
+        }
+      },
+    });
+
+    expect(fragment.querySelector('figure')).toBeTruthy();
+    expect(fragment.querySelector('figcaption')).toBeTruthy();
+    expect(fragment.querySelector('figcaption')?.textContent).toBe('A test image');
+    expect(fragment.querySelector('img')).toBeTruthy();
+    expect(fragment.querySelector('img')?.getAttribute('loading')).toBe('lazy');
+    expect(fragment.querySelector('img')?.src).toBe('https://example.com/');
+    expect(fragment.querySelector('img')?.alt).toBe('A test image');
+    expect(fragment.querySelector('img')?.style.aspectRatio).toBe('150/150');
```

Comment:
Good point!

Reply:
I have consulted the mighty ChatGPT and it mentioned some valid points. I've broken down the tests into smaller chunks and grouped logical properties together. ChatGPT mentioned that you don't want to make the tests too specific, because it will be more difficult to maintain. Testing in isolation also leads to redundant tests and you would be wasting performance because of that.

I think what I have now seems like a good way of testing

Reply:
I think this is a good balance indeed. If pressed I would say that the loading attribute is still better suited as a separate test, but you don't have to change it.